bash/asm

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# asm: run macro assembly to generate relocatible obj from mlc source 
+
+# debug flag; 0=no debug, 1=debug
+debug=0
+
+ECHO=0
+if [ "$1" = "tron" ] || [ "$1" = "TRON" ]; then
+#    echo "setting echo to 1"
+	ECHO=1
+	shift
+fi
+
+if [ -f "$1.BAL" ]; then rm $1.BAL; fi
+if [ -f "$1.PRN" ]; then rm $1.PRN; fi
+if [ -f "$1.OBJ" ]; then rm $1.OBJ; fi
+if [ -f "$1.ERR" ]; then rm $1.ERR; fi
+if [ -f "$1.STA" ]; then rm $1.STA; fi
+#if [ -f $1.TR* ]; then rm $1.TR*; fi
+shopt -s nullglob
+for f in $1.TR*; do
+#    echo "remove file: $f"
+    if [ -f "$f" ]; then rm "$f"; fi
+done
+shopt -u nullglob
+
+# extract longest substring that ends with "/"
+dir=${0%/*}
+
+# get the z390 directory
+zdir=$(dirname $0)
+zdir=$(cd $zdir && pwd)
+zdir=$(dirname $zdir)
+
+sysmac='sysmac(+'$zdir'/mac)'
+syscpy='syscpy(+'$zdir'/mac)'
+
+${dir}/mz390 $1 $sysmac $syscpy $2 $3 $4 $5 $6 $7 $8 $9
```

Comment:
Is it worth moving to the bash way of passing the remaining items to the next command.
* `"$@"` provides all the parameters passed on the command line.. 
* `shift` command remove the variables in the list.

Means you can have more than 2-9 options passed.
Would look something like:

    module=$1
    shift
    mz390 $module $sysmac $syscpy "$@"

Reply:
I'd forgotten about $@. Why the double quotes enclosing $@ ?

Reply:
I am not 100% why.. but all the guides I read said you should.. I think it might be to ensure that bash expands it.

Reply:
While I think using $@ is very cool, I would prefer to leave the scripts as-is. Since the bash scripts are new, any one who wants to compare a bat file to its corrsponding bash script will see similar code.